### PR TITLE
LE audio: CAP common functionality 

### PR DESCRIPTION
**Pull request by Thalley**
Move some code from cap_initiator.c to cap_common.c as it will be shared between the CAP Initiator and the CAP Commander.

### CODE REVIEW

File: subsys/bluetooth/audio/cap_common.c [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+
+#include "cap_internal.h"
+#include "csip_internal.h"
+
+/* Use the highest debug level of the two */
+#if CONFIG_BT_CAP_INITIATOR_LOG_LEVEL > CONFIG_BT_CAP_COMMANDER_LOG_LEVEL
```
Comment (larsgk) [inline subsys/bluetooth/audio/cap_common.c:13]: can one be undefined?

Comment (Thalley) [reply subsys/bluetooth/audio/cap_common.c:13]: Good point

File: subsys/bluetooth/audio/cap_common.c [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+LOG_MODULE_REGISTER(bt_cap_common, CONFIG_BT_CAP_INITIATOR_LOG_LEVEL);
+#else
+LOG_MODULE_REGISTER(bt_cap_common, CONFIG_BT_CAP_COMMANDER_LOG_LEVEL);
+#endif /* CONFIG_BT_CAP_INITIATOR_LOG_LEVEL > CONFIG_BT_CAP_COMMANDER_LOG_LEVEL */
+#elif defined(CONFIG_BT_CAP_INITIATOR)
+LOG_MODULE_REGISTER(bt_cap_common, CONFIG_BT_CAP_INITIATOR_LOG_LEVEL);
+#elif defined(CONFIG_BT_CAP_COMMANDER)
+LOG_MODULE_REGISTER(bt_cap_common, CONFIG_BT_CAP_COMMANDER_LOG_LEVEL);
+#endif /* CONFIG_BT_CAP_INITIATOR && CONFIG_BT_CAP_COMMANDER */
```
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/cap_common.c:23]: This looks odd. Why don't you define a separate Kconfig logging option for this file?

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/cap_common.c (hunk 2 above)
Comment (Thalley) [reply subsys/bluetooth/audio/cap_common.c]: Added a CAP common log level

**Review by MariuszSkamra — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**